phy/mod_ifdefs.F90

```diff
@@ -0,0 +1,22 @@
+module mod_ifdefs
```

Comment:
I this file needed? BLOM ifdefs aren't touched in this PR?

Reply:
You still need this file - since these are the #ifdefs outside of HAMOCC that I have not touched yet.
I would be happy to tackle this - but a lot of the files where these appear are not .F90 files and also not modules.
I'd like to see that happen - but it seemed too invasive for this PR.